進捗分まで

Takuho3650 opened the pull request:
<!--
Pull Requestありがとうございます！
Pull Requestを作成するにあたり、以下の内容を記述してください。

Tip: ファイルをドラッグ＆ドロップすると、画像やファイルを添付できます。
-->

## ⭐️ 概要

<!--
このPull Requestで何を行なったのか、
どう変わるのかを明確かつ簡潔に記述してください。
-->

- DRFの環境構築
- データベースの作成
- API機能の作成
1. GET `ドメイン/api/v1/voter?user_id=<int>`
2. POST `ドメイン/api/v1/voter/`
3. POST `ドメイン/api/v1/vote/`
4. 各種スタッフ用のAPI `ドメイン/api/v1/` を参照

## 😎 目的

<!--
このPull Requestでなぜその変更を行なったのか、
どのような意図なのかを明確かつ簡潔に記述してください。
-->

## 📷 スクリーンショット

<!--
スクリーンショットがあれば添付してください。
-->

## 💬 関連する Issue

<!--
関連するIssueがあれば番号を記述してください。
-->

## 🖋️ 補足事項

<!--
相談したい内容や残っている課題など、
伝えたい内容があれば記述してください。
また、関連するリンクや参考資料などがあれば教えてください。
-->

現在残ってる課題
- APIのテスト(単純なケースでしか試してません)
- 集計用のAPI構築

Takuho3650 committed:
first initial commit

Django環境の構築

Takuho3650 committed:
reflect progress

進捗分まで
・Voters(GET・POST)の機能
・Vote(POST)の機能
・各種スタッフ用の機能(各テーブルのGET・POST・PUT・DELETE)

Takuho3650 committed:
migrate

Takuho3650 commented:
pull

aoba-s commented on api/apis.py:
```diff
@@ -0,0 +1,94 @@
+class StuffLatestVotesViewSet(viewsets.ModelViewSet):
+    serializer_class = LatestVotes_serializer
+    queryset = Latest_votes.objects.all()
+    permission_classes = [IsAuthenticated]
```
[imo]
クラス名、Stuff→Staffかも？

aoba-s commented on api/serializers.py:
```diff
@@ -0,0 +1,51 @@
+class Plans_serializer(serializers.ModelSerializer):
+    class Meta:
+        model = Plans
+        fields = '__all__'
```
[fyi]
ファイル末尾に⛔️が表示されるやつ、この設定すると出なくなるよ（自分も最近知った）

https://qiita.com/hamacccccchan/items/11c17c7412a5aeb2ad74

aoba-s commented on config/settings.py:
```diff
@@ -0,0 +1,152 @@
+# SECURITY WARNING: keep the secret key used in production secret!
+
+# SECRET_KEY = env('SECRET_KEY')
+SECRET_KEY = 'django-insecure-xcx(&9*(=hdb671zs7+^hhft@90(823^8&93+mlr^++%@#kegi'
```
[must]
.envに書いたほうがいいかも

aoba-s commented on api/urls.py:
```diff
@@ -0,0 +1,16 @@
+urlpatterns = [
+    path("api/", include(router.urls)),
+    path("api/voters/<int:pk>/", apis.VotersView.as_view(), name="voters"),
+    path('api/vote/', apis.VoteView.as_view(), name="vote"),
```
[imo]
api/**v1**/xxxかも？

aoba-s commented on api/models.py:
```diff
@@ -0,0 +1,30 @@
+    # example leaflet 3page 15th Plan ==> 315
+    # example leaflet 30page 5th Plan ==> 3005
+    title = models.CharField(max_length=50) # example ARイニャー
+    category_id = models.ForeignKey(Category, on_delete=models.CASCADE)
```
[nits]
DjangoのForeignKey、勝手にフィールド名に「_id」って追加されるらしい
`category`でいいかも
https://docs.djangoproject.com/ja/5.1/ref/models/fields/#database-representation

aoba-s commented on api/models.py:
```diff
@@ -0,0 +1,30 @@
+    user_id = models.IntegerField(primary_key=True)
+
+class Category(models.Model):
+    category_id = models.IntegerField(primary_key=True)
```
[nits]
カテゴリのID、自動採番で良さそうならAutoFieldとかBigAutoFieldって書くといいっぽい
https://github.com/AKABANEDAI-FES/2024-contest/blob/c1aed5021eda57911a3c2ba723d4ec646a8fba3a/api/migrations/0001_initial.py#L39
BigAutoFieldだと統一感があって良いかなぁ？
https://docs.djangoproject.com/ja/5.1/topics/db/models/#automatic-primary-key-fields

aoba-s commented on api/models.py:
```diff
@@ -0,0 +1,30 @@
+from django.db import models
+
+class Voters(models.Model):
```
[imho]
Djangoのクラス名は単数形にするのが良いらしい（？）
https://dottrail.codemountains.org/py-django-model/

aoba-s reviewed (commented):
view.pyだけ詳しく見れてないんだけど他は見てコメントしたので確認してほしい！

Takuho3650 committed:
修正

・Stuff→Staff
・最終行を改行
・シークレットキーをenvに移動
・URLを正しいものに修正
・カラムの命名規則を統一
・自動で問題ないものをBigAutoFieldに（VoterはQRと連動のためIntegerのまま）
・命名を単数形に統一

aoba-s commented on api/models.py:
```diff
@@ -1,30 +1,30 @@
-class Plans(models.Model):
-    plan_id = models.IntegerField(primary_key=True)
+class Plan(models.Model):
+    plan_id = models.BigAutoField(primary_key=True)
```
[imo]
企画IDは手動で設定するだろうからIntegerFieldでいいかも？

Takuho3650 replied:
あそうだった（）
修正しときます！！！

Takuho3650 commented:
書き漏らし
・voterのGETとPOSTをpkなしで処理出来るように

aoba-s reviewed (approved):
一点だけコメントしました！
他は良きかなと思います、ありがとう！
![LGTM](https://image.lgtmoon.dev/117191)

Takuho3650 committed:
Planの修正

BigAutoField→IntegerField
